qiskit_experiments/calibration/experiments/spectroscopy.py

```diff
@@ -0,0 +1,350 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class."""
+
+from typing import List, Optional, Tuple, Union
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate, Parameter
+from qiskit.exceptions import QiskitError
+import qiskit.pulse as pulse
+from qiskit.qobj.utils import MeasLevel
+from qiskit_experiments.analysis.curve_fitting import curve_fit
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments import AnalysisResult
+from qiskit_experiments.data_processing.data_processor import DataProcessor
+from qiskit_experiments.data_processing.nodes import ToReal
+from qiskit_experiments.data_processing.nodes import Probability
+from qiskit_experiments.analysis.plotting import plot_curve_fit, plot_scatter
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class SpectroscopyAnalysis(BaseAnalysis):
+    """Class to analysis a spectroscopy experiment."""
+
+    # pylint: disable=arguments-differ
+    def _run_analysis(
+        self,
+        experiment_data,
+        data_processor: Optional[callable] = None,
+        meas_level: Optional[int] = MeasLevel.CLASSIFIED,
```

Comment:
Are there situations where the user only has access to classified results? I think kerneled would be a better default. Outside of the initial screening of the device, the most likely motivation for running spectroscopy is to look for features related to other transitions near the qubit. The signals from features other than the qubit might not fit the qubit's discriminator well.